tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationOrigins.cs

```diff
@@ -0,0 +1,50 @@
+﻿// <copyright file="ConfigurationOrigins.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.ComponentModel;
+
+namespace Datadog.Trace.Configuration.Telemetry;
+
+internal enum ConfigurationOrigins
+{
+    /// <summary>
+    /// Configuration that is set through environment variables
+    /// </summary>
+    [Description("env_var")]
+    EnvVars,
+
+    /// <summary>
+    /// Configuration that is set through the customer application
+    /// </summary>
+    [Description("code")]
+    Code,
+
+    /// <summary>
+    /// Configuration that is set by the dd.yaml file or json
+    /// </summary>
+    [Description("dd_config")]
+    DdConfig,
```

Comment:
Both `Dd` and `Config` in this name could apply to any source (they are all "Datadog configuration").

Maybe `File`?